### PR TITLE
Add mdms call for checklist, Integrate fsm update api for rating submission

### PR DESCRIPTION
**Pull request by ayan-egov**

### CODE REVIEW

**Comment by ayan-egov** [comment]
Screenshot from 2021-02-12 11-31-39
Screenshot from 2021-02-12 11-31-43
Screenshot from 2021-02-12 11-31-08

**Review by mahendra-egov — APPROVED** [review]

File: packages/modules/fsm/src/pages/citizen/ApplicationDetails.js [hunk 1]
```diff
@@ -145,6 +145,11 @@ const ApplicationDetails = () => {
             <SubmitBar label={t("CS_APPLICATION_DETAILS_MAKE_PAYMENT")} />
           </Link>
         )}
+        {["COMPLETED", "CANCELED"].includes(application.applicationStatus) && (
```
[the remark this answers is not in the document]
Comment (saurabh-egov) [reply packages/modules/fsm/src/pages/citizen/ApplicationDetails.js:167]: done

File: packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js [hunk 2]
```diff
@@ -1,17 +1,73 @@
...
+import { useHistory, useParams } from "react-router-dom";
 import { useTranslation } from "react-i18next";
 
 const SelectRating = ({ parentRoute }) => {
   const { t } = useTranslation();
+  const tenantId = Digit.ULBService.getCurrentTenantId();
   const history = useHistory();
-  // console.log("parent route", parentRoute);
+  let { id: applicationNos } = useParams();
+  const { isError, error, data: application } = Digit.Hooks.fsm.useSearchAll(tenantId, { applicationNos });
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js:11]: why are we using searchAll?

File: packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js [hunk 3]
```diff
@@ -1,17 +1,73 @@
...
+  const { isLoading, data: checklistData } = Digit.Hooks.fsm.useMDMS(tenantId, "FSM", "Checklist");
+  const mutation = Digit.Hooks.fsm.useApplicationUpdate(tenantId);
+  const [answers, setAnswers] = useState({});
 
-  function log(data) {
-    // history.push(`${parentRoute}/response`);
+  function handleSubmit(data) {
+    const { rating } = data;
+    const allAnswers = { ...data, ...answers };
+    let checklist = Object.keys(allAnswers).reduce((acc, key) => {
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js:19]: why are we using this just watch nect action

File: packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js [hunk 4]
```diff
@@ -1,17 +1,73 @@
...
+      if (key === "comments" || key === "rating") {
+        return acc;
+      }
+      acc.push({ code: key, value: Array.isArray(allAnswers[key]) ? allAnswers[key].join(",") : allAnswers[key] });
+      return acc;
+    }, []);
+
+    const [applicationData] = application;
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js:27]: this should not be done here. dont mediate with services if not needed

File: packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js [hunk 5]
```diff
@@ -1,17 +1,73 @@
...
+    // };
+
+    history.push("/digit-ui/employee/fsm/response", {
+      applicationData,
+      key: "update",
+      action: "SUBMIT_FEEDBACK",
+    });
   }
 
+  const handleSuccess = () => {
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js:48]: why are we logging console on success?

File: packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js [hunk 6]
```diff
@@ -23,30 +79,30 @@ const SelectRating = ({ parentRoute }) => {
...
-        type: "checkbox",
-        label: t("CS_FEEDBACK_WHAT_WENT_WRONG"),
-        checkLabels: [
-          t("CS_FEEDBACK_THERE_WAS_SPILLAGE"),
-          t("CS_FEEDBACK_OPERATORS_WERE_WITHOUT_SAFETY_GEARS"),
-          t("CS_FEEDBACK_RESOLUTION_TIME_WAS_LONG"),
-          t("CS_FEEDBACK_OTHERS"),
-        ],
-      },
+      // {
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/Rating/SelectRating.js:82]: remove this if not needed

**Review by saurabh-egov — APPROVED** [review]

**Pull request merged**